### DISCUSSION TITLE
Native support for Next.js 13 Server Components

### DISCUSSION THREAD

**Discussion by brillaintlcd** [styled-components/styled-components#4122] [Ideas]
I think by using Styled Components with **Next.js** 13 Server Components comes at a cost of losing the features that come with Server components & the simplicity of writing so much less code.

Is there any ongoing work to tackle this issue? I would love to work together on a PR if there is any?

**Comment by quantizor** [comment]
This has been implemented. The upcoming release adds native React Server Component support — styled components can now be used directly in server components without `'use client'`. Styles are emitted as plain inline `<style>` tags in the server component output, with no client-side JavaScript required.

Key details:
- No `'use client'` needed for basic styled components in RSC
- Context-dependent APIs (`ThemeProvider`, `StyleSheetManager`) still require client components
- Base CSS in inheritance chains is wrapped in `:where()` for zero specificity to handle cross-boundary extension correctly
- No hydration mismatch since server component output is not hydrated by React

You can test it now:

```
npm install styled-components@test
```

Staged release PR: https://github.com/styled-components/styled-components/pull/5687
